### DISCUSSION TITLE
custom data type

### DISCUSSION THREAD

**Discussion by esisl** [pimcore/pimcore#7811] [Q&A / Support] [answered]
I need a custom data type (more precisely, an editor).
I created files:
/pimcore/bundles/AdminBundle/Resources/public/js/pimcore/object/tags/myType.js
/pimcore/bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/myType.js
/pimcore/models/DataObject/ClassDefinition/Data/myType.php

Everything is fine, a new data type has appeared, it is safely added to the class and edited in the object, like a text field.

But...
1) I did it all "by hand", just creating and editing files. "I think I'm doing something wrong" (c) :)
2) I need to make a rather complex interface, which I would like to take from the /public/js/pimcore/object/tags/myType.js file into a separate file. How to do it? I can't just write "<script src =" ... "> </script>" somewhere
I can add the file by hand to /pimcore/bundles/AdminBundle/Resources/views/Admin/Index/index.html.php
But I suspect that this will be CATEGORALLY wrong.

So what should I ask?

P.S. There is no "Create new plugin skeleton" button in the admin interface

**Comment by pradeep-pimcore** [comment, marked as answer]
Is this helpful
https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Bundle_Developers_Guide/Adding_Object_Datatypes.html

For creating a new plugin/bundle with skeleton you can use _bin/console pimcore:generate:bundle_ . After that you can create the files ( which are mentioned in the above link) in new separate bundle.
